src/gosling-track/gosling-track.ts

```diff
@@ -89,47 +90,77 @@ function initProcessedTileInfo(): ProcessedTileInfo {
 
+const factory: PluginTrackFactory<Tile, GoslingTrackOptions> = (HGC, context, options) => {
     // Services
     const { tileProxy } = HGC.services;
 
-    class GoslingTrackClass extends HGC.tracks.BarTrack {
-        private viewUid: string;
-        private options!: GoslingTrackOption;
+    /* Custom loading label */
+    const loadingTextStyle = getTextStyle({ color: 'black', size: 12 });
+
+    class GoslingTrackClass extends HGC.tracks.BarTrack<Tile, typeof options> {
+        // This is tracking the xScale of an entire view, which is used when no tiling concepts are used
+        drawnAtScale = HGC.libraries.d3Scale.scaleLinear();
+        scalableGraphics: Record<string, PIXI.Graphics> = {};
+        extent = { min: Number.MAX_SAFE_INTEGER, max: Number.MIN_SAFE_INTEGER };
+        hideMousePosition?: () => void;
+
         private tileSize: number;
-        private isRangeBrushActivated: boolean;
         private mRangeBrush: LinearBrushModel;
-        private _xScale!: ScaleLinear<number, number>;
-        private _yScale!: ScaleLinear<number, number>;
-        /** left and top offsets of a track */
-        private position!: [number, number];
-        /** width and height of a track */
-        private dimensions!: [number, number];
-        private pMouseHover: Graphics;
-        private pMouseSelection: Graphics;
+
         private assembly?: Assembly;
-        private fetchedTiles: FetchedTiles;
         private processedTileInfo: Record<string, ProcessedTileInfo>;
         // TODO: add members that are used explicitly in the code
-
-        constructor(params: any[]) {
-            const [context, options] = params;
+        //
+        // Brushes on the color legend
+        private gLegend? = HGC.libraries.d3Selection.select(context.svgElement).append('g');
+        private pMouseHover = new HGC.libraries.PIXI.Graphics();
+        private pMouseSelection = new HGC.libraries.PIXI.Graphics();
+        private isRangeBrushActivated = false;
+        private gBrush = HGC.libraries.d3Selection.select(context.svgElement).append('g');
+
+        private loadingTextStyleObj = new HGC.libraries.PIXI.TextStyle(loadingTextStyle);
+        private loadingTextBg = new HGC.libraries.PIXI.Graphics();
+        private loadingText = new HGC.libraries.PIXI.Text('', loadingTextStyle);
+
+        private svgData: unknown[] = [];
+        private textGraphics: unknown[] = [];
+        /** used to improve the performance of text rendering */
+        private textsBeingUsed = 0;
+        private loadingStatus = { loading: 0, processing: 0, rendering: 0 };
+
+        private prevVisibleAndFetchedTiles?: Tile[];
+        private oldDimensions?: [number, number];
+        private xDomain?: number[];
+        private xRange?: number[];
+
+        private mouseDownX = 0;
+        private mouseDownY = 0;
+
+        constructor() {
             super(context, options);
 
             context.dataFetcher.track = this;
-            this.context = context;
```

Comment:
We have `context` available in the closure. I don't think adding even more to the class hierarchy is very useful.